### PR TITLE
support document id

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

**Comment by OsmondJiang** [comment]
#2821

File: src/docfx/config/ConfigExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+using System;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class ConfigExtensions
+    {
+        public static (string depotName, string docsetName) SplitName(this Config config)
+        {
+            var nameParts = config.Name.Split(new[] { '.' }, StringSplitOptions.RemoveEmptyEntries);
```
Comment (superyyrrzz) [inline src/docfx/config/ConfigExtensions.cs]: >Split [](start = 40, length = 5)

use string.IndexOf as docsetName may contain `.`

File: docs/specs/markdown.yml [hunk 2]
```diff
@@ -89,9 +88,44 @@ inputs:
     <style>h1 {color:red;}</style>
 
     body
-  build.manifest:
 outputs:
   docs/a.json: |
     { "content": "<div></div><p>body</p>" }
   build.manifest:
-
+---
```
[the remark this answers is not in the document]
Comment (OsmondJiang) [reply docs/specs/markdown.yml]: i will put them to metadata.yml

---
In reply to: [196295315](https://github.com/dotnet/docfx/pull/2895#discussion_r196295315) [](ancestors = 196295315)

File: src/docfx/config/ConfigExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class ConfigExtensions
+    {
+        public static (string depotName, string docsetName) SplitName(this Config config)
```
[the remark this answers is not in the document]
Comment (OsmondJiang) [reply src/docfx/config/ConfigExtensions.cs]: it's backward compatibility filed, i want to put it in a extension

---
In reply to: [196296662](https://github.com/dotnet/docfx/pull/2895#discussion_r196296662) [](ancestors = 196296662)

**Pull request merged**